src/vos/vos_tree.c

```diff
@@ -936,15 +976,12 @@ key_tree_prepare(struct vos_object *obj, daos_handle_t toh,
 	case 0:
 		krec = rbund.rb_krec;
 		ilog = &krec->kr_ilog;
-		found = vos_ilog_ts_lookup(ts_set, ilog);
-		if (found)
-			break;
 		/** fall through to cache re-cache entry */
 	case -DER_NONEXIST:
 		/** Key hash already be calculated by dbtree_fetch so no need
 		 *  to pass in the key here.
 		 */
-		tmprc = vos_ilog_ts_cache(ts_set, ilog, NULL, 0);
+		tmprc = vos_ilog_ts_add(ts_set, ilog, NULL, 0);
```

Comment:
It's to avoid rehash of the key that we just hashed.   We save the hashed value and use it just for this one.   The next call to vos_ilog_ts_add will either pass arguments or will have replaced the stored hash.